### ISSUE TITLE
The implementation of EfficientNet in keras.applications is inconsistent

### ISSUE THREAD

**Issue by aingo03304** [keras-team/keras#14950]
**Describe the problem**.

In `keras.applications`, EfficientNet starts with two layers, scaling and normalization.  
However, other models do not have such layers.  
I don't think it is deliberately designed. So, can you check on this?

https://github.com/keras-team/keras/blob/d369e05c1075bb6732537d1a69d1fd4f5d0aa749/keras/applications/efficientnet.py#L319-L322

What is worse is that mixed-precision training does not support `layers.Normalization`.
Because this is not a problem related to this issue, I am not going to report this specifically here, but make another issue.

**Describe the current behavior**.
None, this is not a functional error.

**Describe the expected behavior**.
None, this is not a functional error.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

- Do you want to contribute a PR? (yes/no): yes
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing): If the implementation is against the design choice, I would like to delete two lines containing scaling and normalization.  

**Standalone code to reproduce the issue**.

Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

**Source code / logs**.

Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by jvishnuvardhan** [comment]
> However, other models do not have such layers.

@aingo03304 Can you please describe with little more details on the inconsistency and what other models you are referring here? Thanks!

**Comment by aingo03304** [comment]
@jvishnuvardhan Thank you for the quick response.

For example, ResNet does not scale and normalize the input tensor.
https://github.com/keras-team/keras/blob/d369e05c1075bb6732537d1a69d1fd4f5d0aa749/keras/applications/resnet.py#L143-L155

In MobileNetV2, there is a note that each Keras application model requires each specific shape.
https://github.com/keras-team/keras/blob/d369e05c1075bb6732537d1a69d1fd4f5d0aa749/keras/applications/mobilenet_v2.py#L129-L132

However, EfficientNet implementation contains scaling and normalization alone.  
As far as I know, EfficientNet is the only implementation that contains data preprocessing layers.  
I think this can confuse users. Any thought on this?

**Comment by aingo03304** [comment]
In addition, with the code below, I got an error if I turn on `--model EfficientNetB0`.

```python
import os
import argparse
import tensorflow as tf
import tensorflow_datasets as tfds
import tensorflow.keras.mixed_precision.experimental as mixed_precision

model_name = [
    'ResNet50',
    'ResNet101',
    'ResNet152',
    'VGG16',
    'VGG19',
    'EfficientNetB0',
    'EfficientNetB1',
    'EfficientNetB2',
    'EfficientNetB3',
    'EfficientNetB4',
    'EfficientNetB5',
    'EfficientNetB6',
    'EfficientNetB7',
]

parser = argparse.ArgumentParser()
parser.add_argument('--learning-rate', type=float, default=0.1, help="LR for SGD")
parser.add_argument('--momentum', type=float, default=0.9, help="Momentum for SGD")
parser.add_argument('--batch-size', type=int, default=32, help="Batch size")
parser.add_argument('--epochs', type=int, default=10, help="# of epoch")
parser.add_argument('--model', type=str, default='ResNet50', help="model name")
parser.add_argument('--datadir', type=str, required=True, help="dataset dir")
parser.add_argument('--num-workers', type=int, default=1, help="dataloader #threads")
args = parser.parse_args()

IMG_DIM = 224
policy = mixed_precision.Policy('mixed_float16')
mixed_precision.set_policy(policy)

print(f'Compute dtype: {policy.compute_dtype}')
print(f'Variable dtype: {policy.variable_dtype}')

def get_wrapped_model():
    input_shape = (IMG_DIM, IMG_DIM, 3)
    image_input = tf.keras.layers.Input(shape=input_shape)
    model_fn = getattr(tf.keras.applications, args.model)
    model = model_fn(
        include_top=True,
        weights=None,
        input_tensor=image_input,
        pooling='max',
        classes=1000)
print("Loading datasets... This takes some time.")
train_dataset = tf.keras.preprocessing.image_dataset_from_directory(
    os.path.join(args.datadir, 'train'),
    labels="inferred",
    label_mode="int",
    class_names=None,
    color_mode="rgb",
    batch_size=args.batch_size,
    image_size=(IMG_DIM, IMG_DIM),
    shuffle=True,
    seed=42,
    validation_split=None,
    subset=None,
    interpolation="bilinear",
    follow_links=False,
)

model = get_wrapped_model()
model.compile(
    optimizer='sgd',
    loss='sparse_categorical_crossentropy',
    metrics=['accuracy'])

model.fit(train_dataset)
```


```
Traceback (most recent call last):
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/ops/math_ops.py", line 1159, in binary_op_wrapper
    out = r_op(x)
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/ops/math_ops.py", line 1178, in r_binary_op_wrapper
    x = ops.convert_to_tensor(x, dtype=y.dtype.base_dtype, name="x")
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/profiler/trace.py", line 163, in wrapped
    return func(*args, **kwargs)
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/framework/ops.py", line 1511, in convert_to_tensor
    raise ValueError(
ValueError: Tensor conversion requested dtype float16 for Tensor with dtype float32: <tf.Tensor 'normalization/Cast:0' shape=(None, 224, 224, 3) dtype=float32>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "main.py", line 73, in <module>
    model = get_wrapped_model()
  File "main.py", line 45, in get_wrapped_model
    model = model_fn(
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/keras/applications/efficientnet.py", line 531, in EfficientNetB0
    return EfficientNet(
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/keras/applications/efficientnet.py", line 317, in EfficientNet
    x = layers.Normalization(axis=bn_axis)(x)
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py", line 971, in __call__
    return self._functional_construction_call(inputs, args, kwargs,
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py", line 1110, in _functional_construction_call
    outputs = self._keras_tensor_symbolic_call(
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py", line 841, in _keras_tensor_symbolic_call
    return self._infer_output_signature(inputs, args, kwargs, input_masks)
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py", line 882, in _infer_output_signature
    outputs = call_fn(inputs, *args, **kwargs)
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/keras/layers/preprocessing/normalization.py", line 206, in call
    return ((inputs - mean) /
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/ops/math_ops.py", line 1164, in binary_op_wrapper
    raise e
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/ops/math_ops.py", line 1148, in binary_op_wrapper
    return func(x, y, name=name)
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/util/dispatch.py", line 206, in wrapper
    return target(*args, **kwargs)
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/ops/math_ops.py", line 539, in subtract
    return gen_math_ops.sub(x, y, name)
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/ops/gen_math_ops.py", line 10565, in sub
    _, _, _op, _outputs = _op_def_library._apply_op_helper(
  File "/home/aingo03304/anaconda3/envs/xla/lib/python3.8/site-packages/tensorflow/python/framework/op_def_library.py", line 555, in _apply_op_helper
    raise TypeError(
TypeError: Input 'y' of 'Sub' Op has type float16 that does not match type float32 of argument 'x'.
```

**Comment by reedwm** [comment]
> What is worse is that mixed-precision training does not support `layers.Normalization`.

This should be fixed in TensorFlow 2.6, which is out now.
